lib/utils/helpers.ts

```diff
@@ -92,6 +92,13 @@ export async function getManifest(
 	return balena.models.device.getManifestBySlug(deviceType);
 }
 
+export const areDeviceTypesCompatible = (
+	deviceTypeA: BalenaSdk.DeviceType,
+	deviceTypeB: BalenaSdk.DeviceType,
+) =>
+	deviceTypeA.arch === deviceTypeB.arch &&
+	!!deviceTypeA.isDependent === !!deviceTypeB.isDependent;
```

Comment:
Why do you need to check for "isDependent"?

Reply:
Because you can't/shouldn't configure a dependent device type for a top level app, even if they are of the same arch.

Reply:
Then the check you need should be `!deviceTypeA.isDependent === !deviceTypeB.isDependent` right?

Reply:
The typings have it as nullable, and it's sometimes missing depending on the config.json on s3, so I decided it to normalize the boolean values, like we ended up doing in the UI.

Reply:
`!!deviceTypeA.isDependent === !!deviceTypeB.isDependent` indeed ensures you'll compare 2 booleans but also means that when both are `true` the function will return `true` (i.e. "yes, the device types are compatible") which isn't what you describe about dependent device types above.

Reply:
Oh, I just realized that you are just skipping the duplicate inverse checks :)
I would say let's keep it as is since `!!x` is an easier patter to spot conversion to boolean, and js engines optimize that anyway.

Reply:
That's not my point :) I'm saying that with `!!` this function will return "true" both when the device types are of dependent devices *and* when they're not, but if I understand your comment above correctly the latter shouldn't be allowed, in which case you need to ensure *both* are *not dependent*. Makes sense?

Reply:
Discussed on FD.
See: https://www.flowdock.com/app/rulemotion/resin-tech/threads/7rPD_QLlNzK9uobomhHPgruDgba